### PR TITLE
Add new status endpoint for a single file

### PR DESCRIPTION
**Pull request by mryan321**
### What

This PR adds a new status endpoint for returning status of a single uploaded file. More details can be found here: https://trello.com/c/zJMQlrOY

Its not as big as it looks - it can be summed up in:
- a new endpoint: https://github.com/ONSdigital/dp-upload-service/blob/7a1d50f41315dfed7c2cc4a204843013067600a7/service/service.go#L92
- a new handler: https://github.com/ONSdigital/dp-upload-service/blob/add-status-endpoint-for-single-file/api/status.go
- an extension of the files store: https://github.com/ONSdigital/dp-upload-service/blob/7a1d50f41315dfed7c2cc4a204843013067600a7/files/store.go#L62

It also includes a bit of housekeeping - only where relevant - a little around s3 buckets, some minor re-organising for readability, consistent mocks.

Note:
I do feel like this project steers a little against the norm for ONS APIs - I have raised a ticket for looking at better http response handling with https://github.com/ONSdigital/dp-net/blob/main/responder/examples/basic_handler/main.go - also at deprecating the old endpoint. But please let me know if you spot anything else that needs attention. In the whole I have tried to follow consistent development practices taken within this repo.

### How to review

Check tests pass - if you want to run locally please reach out 🤠 

### Who can review

Anyone

### CODE REVIEW

File: files/store.go [hunk 1]
```diff
@@ -7,35 +7,92 @@ import (
...
+	FileContent   StatusMessage      `json:"file_content"`
+}
+
+func NewStore(files FilesClienter, bucket *aws.Bucket, vault *encryption.Vault) Store {
+	return Store{files, bucket, vault}
+}
+
+func (s Store) Status(ctx context.Context, path string) (*Status, error) {
+	//metadata
+	metadata, err := s.files.GetFile(ctx, path, "")
```
Comment (greg-fujar) [inline files/store.go:64]: should the auth token be passed here?

File: files/store.go [hunk 2]
```diff
@@ -7,35 +7,92 @@ import (
...
+	k, err := s.vault.EncryptionKey(ctx, path)
+	encryptionKey := newStatusMessage(len(k) > 0, err)
+
+	//file content
+	url, err := s.bucket.GetS3URL(path)
+	if err != nil {
+		log.Error(ctx, "failed to get file content", err, log.Data{"path": path})
+		return nil, ErrS3Download
+	}
+	//todo encryption off??? handle this with other Get call
```
Comment (greg-fujar) [inline files/store.go:80]: it looks like it falls back to unencrypted version automatically if the psk slice is nil: https://github.com/ONSdigital/dp-s3/blob/f8b5427c664db3856d378d66c9c2fc09475ada7a/get.go#L70

File: files/store.go [hunk 3]
```diff
@@ -7,35 +7,92 @@ import (
...
+	_, size, err := s.bucket.GetFromS3URLWithPSK(url, aws.PathStyle, k)
```
Comment (greg-fujar) [inline files/store.go:81]: given we're swallowing the body, would it make sense to make a Head call instead?

File: upload/upload.go [hunk 4]
```diff
@@ -42,19 +43,17 @@ func (resum *Resumable) createS3Request() *s3client.UploadPartRequest {
 
 // Uploader represents the necessary configuration for uploading a file
 type Uploader struct {
-	s3Client S3Clienter
+	bucket   *aws.Bucket
 	vault    *encryption.Vault
 	s3Region string
 	s3Bucket string
```
Comment (greg-fujar) [inline upload/upload.go:49]: I think these 2 could be removed now?

File: files/store.go (hunk 3 above)
Comment (mryan321) [reply files/store.go:81]: def!!! changing 👍

File: files/store.go (hunk 2 above)
Comment (mryan321) [reply files/store.go:80]: changing to Head anyway

File: files/store.go (hunk 1 above)
Comment (mryan321) [reply files/store.go:64]: we have an issue here - i cant route via dp-api-router (then to files-api) - it loses the auth token... so can only point direct. we need to fix this I think as sep PR.

i already pass the auth token on client creation too: https://github.com/ONSdigital/dp-api-clients-go/blob/9f50972f822e604f7959b82e0d12a891a52f55c2/files/client.go#L53 - we should change the client Get method to use this.

Comment (mryan321) [reply files/store.go:64]: im not really sure how these new auth permissions work too - need to look into that: https://github.com/ONSdigital/dp-files-api/blob/5953bf9cef81dbe4cb645a993549e3f0a44012cc/service/service.go#L55

**Review by greg-fujar — APPROVED** [review]

**Pull request merged**